README.md

```diff
@@ -1,17 +1,94 @@
 ## Firefox Application Services
 
-### Contents
+_A platform for building cloud-powered applications that target Firefox users_
 
-* [fxa-client](components/fxa-client) - cross compiled FxA Rust client that can work with Firefox Sync keys and more
-* [sandvich-ios](samples/sandvich-ios) - Example apps that use SDKs built on top of `fxa-client` to demonstrate a FxA login flow.
-* [sync15](components/sync15) - Sync 1.5 adapter
-* [libs](libs) - libs directory has build scripts for native libraries
-* [docs](docs) - documentation sources 
-* [website](website) - website built from documentation sources
+### What's this all about?
+
+This repository hosts the code and docs
+needed to integrate with the products offered by
+the Firefox [Application Services](https://mana.mozilla.org/wiki/display/CLOUDSERVICES/Application+Services+Home) team.
+
+If you're interested in getting involved
+in the development of those products
+then you're in the right place!
+If that's not why you're here
+then you might enjoy:
+
+* The [Application Services Product Portal](https://mozilla.github.io/application-services/),
+  if you're looking to use those products in your application.
+* The [Application Services Team Home](https://mana.mozilla.org/wiki/display/CLOUDSERVICES/Application+Services+Home)
+  on Mana, if you're trying to find out more about how we build them.
 
 
 ### Overview
 
-The following diagram describes how some of the components hosted here relate to each other within an app.
+This repository is used 
+to build client-side libraries
+for integrating with Firefox Application services
+such as Firefox Accounts, Firefox Sync and Push.
+Each of these is called a "component"
+and is built using a core of shared code
+writen in Rust, exposed over a C FFI
+and wrapped in native language bindings
+for different platforms.
+
+The end result
+is an application
+that can be assembled
+from re-usable components
+that are largely shared across platforms,
+like this:
 
 <img src="https://www.lucidchart.com/publicSegments/view/99d1529a-585a-4f43-bfe8-26ba82f3db51/image.png" width="500" />
+
+The code for these components
+is organized as follows:
+
+* [./libs/](libs) contains infratructure
+  for building some native dependencies, such as OpenSSL.
+* [./components/](components) contains the source
+  for each component, and its FFI bindings.
+  * See [./components/fxa-client/](components/fxa-client)
```

Comment:
As mentioned, I wouldn't use fxa-client for an FFI example, since it does things in a kind of weird way (better perf, harder to maintain) that we probably wouldn't repeat.

Logins is the most complete example that has ios bindings (and doesn't have the weird threading concerns that places is developing).